core/ir/aarch64/encode.c

```diff
@@ -198,6 +199,80 @@ const reg_id_t dr_reg_fixer[] = { REG_NULL,
 };
 /* clang-format on */
 
+/* Maps containing-registers to their DR_ISA_REGDEPS virtual-register.
+ * Same size as dr_reg_fixer[], keep them synched.
+ * Note: DR_REG_INVALID (== DR_REG_V1 == 1) has been excluded to avoid failures on opnd_t
+ * operations.
+ */
+const reg_id_t dr_reg_virtual[] = {
+    DR_REG_V0,   DR_REG_V0,   DR_REG_V2,   DR_REG_V3,   DR_REG_V4,   DR_REG_V5,
```

Comment:
One could say that DR_REG_V0 is the virtual DR_REG_NULL.
For regdeps ISA we don't really need the concept of "absence of a register", which, I think, is what DR_REG_NULL represents.
It's also weird IMHO to have a map from real registers to virtual registers that maps back to a "real" DR_REG_ register rather than a DR_REG_V register.
Note that DR_REG_V0 here would only appear where dr_reg_fixer[] has a DR_REG_NULL, it wouldn't be used for its other "valid" registers.
That being said, we could have a properly named "DR_REG_VNULL" value, rather than a generic "DR_REG_V0", but do we need a "null register" in the realm of regdeps ISA?

Reply:
No more relevant.

Reply:
I think any DR_REG_ enum should leave the sentinel DR_REG_NULL and DR_REG_INVALID intact and not map something else to those values, as too much code acts on those (just as you already found for INVALID).